pennylane/pauli/utils.py

```diff
@@ -1379,24 +1379,23 @@ def _pauli_mult(p1, p2):
     """
     c = 1.0
 
-    t1 = [t[0] for t in p1]
-    t2 = [t[0] for t in p2]
-
+    t1 = np.array([t[0] for t in p1], dtype=int).reshape((1, -1))
+    t2 = np.array([t[0] for t in p2], dtype=int).reshape((-1, 1))
+    d = (t1 - t2) != 0
+    t1notint2 = np.all(d, axis=0)
+    t2notint1 = np.all(d, axis=1)
```

Comment:
Seems that these new changes slightly decrease the performance?

Reply:
Hi Soran, could you please send me the script you use to benchmark? I would like to test it on my end (+ for future use), and then we can compare results.

Reply:
You are right. I think some processes running in parallel on my laptop messed up my numbers. I'll just revert and let's merge the changes as is for now. As you said, `_pauli_mult` can be dealt with later.